src/lib/transport/ip/socket.c

```diff
@@ -50,8 +50,11 @@ void ci_sock_cmn_init(ci_netif* ni, ci_sock_cmn* s, int can_poison)
 {
   /* Poison. */
   CI_DEBUG(
-  if( can_poison )
-    memset(&s->b + 1, 0xf0, (char*) (s + 1) - (char*) (&s->b + 1));
+  if( can_poison ) {
+    citp_waitable b_tmp = s->b;
+    memset(s, 0xf0, sizeof(*s));
+    s->b = b_tmp;
+  }
```

Comment:
Can it be done in the same way as in `efab_ip_queue_copy()` above?

Reply:
done